src/main/scala/com/fulcrumgenomics/vcf/DownsampleVcf.scala

```diff
@@ -0,0 +1,281 @@
+  class DownsampleVcf
+  (@arg(flag = 'i', doc = "The vcf to downsample.") val input: PathToVcf,
+   @arg(flag = 'm', doc = "Index file with bases per sample.") val metadata: Option[FilePath] = None,
+   @arg(flag = 'b', doc = "Original number of bases (for single-sample VCF)") val originalBases: Option[Double] = None,
+   @arg(flag = 'n', doc = "Target number of bases to downsample to.") val downsampleToBases: Double,
+   @arg(flag = 'o', doc = "Output file name.") val output: PathToVcf,
+   @arg(flag = 'w', doc = "Winnowing window size.") val windowSize: Int = 150,
+   @arg(flag = 'e', doc = "Error rate for genotyping.") val epsilon: Double = 0.01,
+   @arg(flag = 'c', doc = "True to write out no-calls.") val writeNoCall: Boolean = false)
+    extends FgBioTool {
+    Io.assertReadable(input)
+    Io.assertReadable(metadata)
+    Io.assertCanWriteFile(output)
+    require(downsampleToBases > 0, "target base count must be greater than zero")
+    require(windowSize >= 0, "window size must be greater than or equal to zero")
+    require(0 <= epsilon && epsilon <= 1, "epsilon/error rate must be between 0 and 1")
+    originalBases match {
+      case Some(x) =>
+        require(x > 0, "originalBases must be greater than zero")
+        require(metadata.isEmpty, "Must pass either originalBases (for single-sample VCF) or metadata, not both")
+      case None =>
+        require(metadata.isDefined, "Must pass either originalBases (for single-sample VCF) or metadata, not both")
+    }
+
+    override def execute(): Unit = {
+      val vcf = VcfSource(input)
+      val progress = ProgressLogger(logger, noun = "variants")
+      val proportions = (
+        originalBases match {
+          case Some(x) =>
+            require(vcf.header.samples.length == 1, "--original-bases requires a single-sample VCF")
+            LazyList(vcf.header.samples.head -> math.min(downsampleToBases / x, 1.0))
+          case _ =>
+            Sample.read(metadata.getOrElse(throw new RuntimeException))
+              .filter(s => vcf.header.samples.contains(s.SAMPLE_NAME))
+              .map(sample => sample.SAMPLE_NAME -> math.min(downsampleToBases / sample.BASE_COUNT.toDouble, 1.0))
+        }
+      ).toMap
+      proportions.foreach { case (s, p) => logger.info(f"Downsampling $s with proportion ${p}%.4f") }
+
+      val winnowed = if (windowSize > 0) winnowVariants(vcf.iterator, windowSize = windowSize, dict = vcf.header.dict) else vcf.iterator
+      val outputVcf = VcfWriter(path = output, header = buildOutputHeader(vcf.header))
+
+      val random = new Random(42)
+      winnowed.foreach { v =>
+        val ds = downsampleAndRegenotype(v, proportions = proportions, random = random, epsilon = epsilon)
+        if (writeNoCall) {
+          outputVcf += ds
+          progress.record(ds)
+        }
+        else if (!ds.gts.forall(g => g.isNoCall)) {
+          outputVcf += ds
+          progress.record(ds)
+        }
+      }
+      
+      progress.logLast()
```

Comment:
probably nice to log how many reads were read, how many remained after winnowing, and how many were written.  Consider three progress loggers?